RecoHI/HiTracking/plugins/HIBestVertexProducer.cc

```diff
@@ -19,6 +19,8 @@ HIBestVertexProducer::HIBestVertexProducer
   theMedianVertexCollection(consumes<reco::VertexCollection>(ps.getParameter<edm::InputTag>("medianVertexCollection"))),
   theAdaptiveVertexCollection(consumes<reco::VertexCollection>(ps.getParameter<edm::InputTag>("adaptiveVertexCollection")))
 {
+  theUseFinalAdapativeVertexCollection  = (ps.existsAs<bool>("useFinalAdapativeVertexCollection") ? ps.getParameter<bool>("useFinalAdapativeVertexCollection") : false);
```

Comment:
existsAs should normally not be used.
Please propagate this parameter to relevant configs.
Better yet, please define fillDescriptions method and define the value of this and other parameters there.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp